doc/service-tests.md

```diff
@@ -115,80 +119,100 @@ That's looking good!
 Sometimes if we have a failing test, it is useful to be able to see some logging output to help work out why the test is failing. We can do that by calling `npm run test:services:trace`. Try running
 
 ```
-npm run test:services:trace -- --only=wercker
+npm run test:services:trace -- --only=docsrs
 ```
 
 to run the test with some additional debug output.
 
 ### (4) Writing More Tests
 
-We should write tests cases for valid paths through our code. The Wercker badge supports an optional branch parameter so we'll add a second test for a branch build.
+We should write tests cases for valid paths through our code. The Docs.rs badge supports an optional version parameter so we'll add a second test for a branch build. In this case, we know for sure that the documentation for this older version was successfully built, we specify a string literal instead of a Joi schema for `message`. This narrows down the expectation and gives us a more helpful error message if the test fails.
 
 ```js
-t.create('Build status (with branch)')
-  .get('/build/wercker/go-wercker-api/master.json')
-  .expectBadge({ label: 'build', message: isBuildStatus })
+t.create('Passing docs for version').get('/tokio/0.3.0.json').expectBadge({
+  label: 'docs@0.3.0',
+  message: 'passing',
+  color: 'brightgreen',
+})
 ```
 
 ```
-Server is starting up: http://lib/service-test-runner/cli.js:80/
-  Wercker
-    Build status
-      ✓
-        [ GET /build/wercker/go-wercker-api.json ] (572ms)
-    Build status (with branch)
-      ✓
-        [ GET /build/wercker/go-wercker-api/master.json ] (368ms)
-
-  2 passing (1s)
+Server is starting up: http://localhost:1111/
+  DocsRs
+    [live] Docs with no version specified
+      √
+        [ GET /tokio.json ] (408ms)
+    [live] Passing docs for version
+      √
+        [ GET /tokio/0.3.0.json ] (171ms)
+
+
+  2 passing (2s)
 ```
 
 Once we have multiple tests, sometimes it is useful to run only one test. We can do this using the `--fgrep` argument. For example:
 
 ```
-npm run test:services -- --only="wercker" --fgrep="Build status (with branch)"
+npm run test:services -- --only="docsrs" --fgrep="Passing docs for version"
 ```
 
-Having covered the typical and custom cases, we'll move on to errors. We should include a test for the 'not found' response and also tests for any other custom error handling. The Wercker integration defines a custom error condition for 401 as well as a custom 404 message:
+We can also add a test for a unsuccessful documentation build:
 
-```js
-httpErrors: {
-  401: 'private application not supported',
-  404: 'application not found',
-}
+```j
+t.create('Failing docs for version')
+  .get('/tensorflow/0.16.1.json')
```

Comment:
There's a couple of places here where we are switching between using tensorflow and tokio. Can we either stick to one, or explain more explicitly why we are using different packages in some examples.

Reply:
There you go, tokio also has a couple of versions with failing builds.